src/serac/coefficients/coefficient_extensions.hpp

```diff
@@ -14,41 +14,170 @@
 
 #include <functional>
 #include <memory>
-
+#include <variant>
+#include <vector>
 #include "mfem.hpp"
 
 #include "serac/numerics/expr_template_ops.hpp"
-
+#include "serac/physics/utilities/finite_element_state.hpp"
 /**
  * @brief Functionality that extends current MFEM capabilities
  *
  */
 namespace serac::mfem_ext {
 
+namespace detail {
+// methods for determining index type
+template <typename T>
+struct index_t {
+  using type = std::size_t;
+};
+
+template <typename T>
+struct index_t<mfem::Array<T>> {
+  using type = int;
+};
+
+// methods for determining the size of a container
+template <typename T>
+auto size(T container)
```

Comment:
To avoid making copies this should be changed to a [universal reference](https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers), i.e., `T&& container`.